commit 95f05a784a24b90d42909b07c4c96e888cc89a45 (refs/changes/45/2745/2)
Author: Trent Mick <trentm@gmail.com>
Date:   2017-10-06T16:53:47-07:00 (2 years ago)
    
    ZAPI-802 drop pointless templating of VMAPI's SMF manifest
    Reviewed by: Todd Whiteman <todd.whiteman@joyent.com>
    Approved by: Todd Whiteman <todd.whiteman@joyent.com>

diff --git a/Makefile b/Makefile
index e086f33..a7c34a9 100644
--- a/Makefile
+++ b/Makefile
@@ -47,7 +47,7 @@ JSL_CONF_NODE	 = tools/jsl.node.conf
 JSL_FILES_NODE   = server.js $(JS_FILES)
 JSSTYLE_FILES	 = server.js $(JS_FILES)
 JSSTYLE_FLAGS    = -o indent=4,doxygen,unparenthesized-return=0
-SMF_MANIFESTS_IN	 = smf/manifests/vmapi.xml.in
+SMF_MANIFESTS	 = smf/manifests/vmapi.xml
 
 include ./tools/mk/Makefile.defs
 ifeq ($(shell uname -s),SunOS)
diff --git a/boot/configure.sh b/boot/configure.sh
index 5f8e578..2394362 100755
--- a/boot/configure.sh
+++ b/boot/configure.sh
@@ -7,16 +7,12 @@
 #
 
 #
-# Copyright (c) 2014, Joyent, Inc.
+# Copyright (c) 2017, Joyent, Inc.
 #
 
 export PS4='[\D{%FT%TZ}] ${BASH_SOURCE}:${LINENO}: ${FUNCNAME[0]:+${FUNCNAME[0]}(): }'
 set -o xtrace
-
-echo "Updating SMF manifest"
-$(/opt/local/bin/gsed -i"" -e "s/@@PREFIX@@/\/opt\/smartdc\/vmapi/g" /opt/smartdc/vmapi/smf/manifests/vmapi.xml)
-
-echo "Importing vmapi.xml"
-/usr/sbin/svccfg import /opt/smartdc/vmapi/smf/manifests/vmapi.xml
+set -o errexit
+set -o pipefail
 
 exit 0
diff --git a/boot/setup.sh b/boot/setup.sh
index 64e1697..c960151 100755
--- a/boot/setup.sh
+++ b/boot/setup.sh
@@ -33,6 +33,8 @@ chown -R nobody:nobody /opt/smartdc/vmapi
 echo "" >>/root/.profile
 echo "export PATH=\$PATH:/opt/smartdc/vmapi/build/node/bin:/opt/smartdc/vmapi/node_modules/.bin" >>/root/.profile
 
+/usr/sbin/svccfg import /opt/smartdc/vmapi/smf/manifests/vmapi.xml
+
 echo "Adding log rotation"
 # Log rotation.
 sdc_log_rotation_add config-agent /var/svc/log/*config-agent*.log 1g
diff --git a/smf/manifests/vmapi.xml.in b/smf/manifests/vmapi.xml
similarity index 88%
rename from smf/manifests/vmapi.xml.in
rename to smf/manifests/vmapi.xml
index 8a696d2..8cc477f 100644
--- a/smf/manifests/vmapi.xml.in
+++ b/smf/manifests/vmapi.xml
@@ -7,7 +7,7 @@
 -->
 
 <!--
-    Copyright (c) 2014, Joyent, Inc.
+    Copyright (c) 2017, Joyent, Inc.
 -->
 
 <service_bundle type="manifest" name="vmapi">
@@ -27,12 +27,12 @@
     <exec_method
       type="method"
       name="start"
-      exec="@@NODE@@ --abort_on_uncaught_exception @@PREFIX@@/server.js &amp;"
+      exec="/opt/smartdc/vmapi/build/node/bin/node --abort_on_uncaught_exception /opt/smartdc/vmapi/server.js &amp;"
       timeout_seconds="30">
       <method_context working_directory="/opt/smartdc/vmapi">
         <method_credential user="nobody" group="nobody" privileges="basic,net_privaddr"/>
         <method_environment>
-          <envvar name="PATH" value="/opt/nodejs/bin:/usr/local/bin:/opt/local/bin:/usr/bin:/usr/sbin:/bin"/>
+          <envvar name="PATH" value="/usr/local/bin:/opt/local/bin:/usr/bin:/usr/sbin:/bin"/>
           <envvar name="LD_PRELOAD_32" value="/usr/lib/extendedFILE.so.1" />
         </method_environment>
       </method_context>
